internal/function/blame.go

```diff
@@ -0,0 +1,179 @@
+package function
+
+import (
+	"fmt"
+	"github.com/src-d/gitbase"
+	"github.com/src-d/go-mysql-server/sql"
+	"gopkg.in/src-d/go-git.v4"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+type BlameGenerator struct {
+	commit  *object.Commit
+	fIter   *object.FileIter
+	curLine int
+	curFile *object.File
+	lines   []*git.Line
+}
+
+func NewBlameGenerator(c *object.Commit, f *object.FileIter) (*BlameGenerator, error) {
+	return &BlameGenerator{commit: c, fIter: f, curLine: -1}, nil
+}
+
+func (g *BlameGenerator) loadNewFile() error {
+	var err error
+	g.curFile, err = g.fIter.Next()
+	if err != nil {
+		return err
+	}
+
+	result, err := git.Blame(g.commit, g.curFile.Name)
+	if err != nil {
+		return err
+	}
+	g.lines = result.Lines
+	g.curLine = 0
+	return nil
+}
+
+func (g *BlameGenerator) Next() (interface{}, error) {
+	if g.curLine == -1 || g.curLine >= len(g.lines) {
```

Comment:
if you have 0 lines (no results), then `g.curLine is 0` and `g.lines is 0`, so it will always try to load, and the it may crash here: `l := g.lines[g.curLine]`
am I missing sth.?

Reply:
If there are 0 lines, `g.curLine` (0) will be `>=` than `g.lines` (0) and then it will load the next file which will stop the iteration if there are no more files.

Reply:
my point was - is it possible that `g.fIter.Next()` will return some *non `io.EOF`*, but `git.Blame(g.commit, g.curFile.Name)` will return an empty result?
If yes, I think it will crash in next line (`g.lines[g.curLine]`).

Reply:
I'm not sure you can have 0 changes lines in a blame (since it would return at least an `Author` even with an empty `Text`), but just in case I've added an extra check to continue (or stop) the iteration in `loadNewFile` if there are 0 lines returned.